### DISCUSSION TITLE
Erase button on "Today's Detections"

### DISCUSSION THREAD

**Discussion by calyptura** [mcguirepr89/BirdNET-Pi#877] [Ideas]
Hi birdneters,
Could someone please add the erase record button on the records shown on "Todays Detections"?
This would make it a lot easier to review the detections.
Thank you very much!
Luciano

**Comment by calyptura** [comment]
Another improvement that would help a lot in reviewing the detections would be having an "All detections" window for any given date (like "Today's Detections") when you go on "Recordings" and "Dates". Presently you need to go checking clicking species by species. 

Thank you!

**Reply by Astrobirder** [reply]
So, I'm not sure if this is related to the above or not, but I've been doing playback to do testing of my newly (re)set-up system, so I don't really want this data in my database long term.  I did a Tools->System Controls->Clear ALL Data and when it had finished and I got back to the overview page I still had this:
![BirdNetPostClearALLDataOverviewScreen](https://user-images.githubusercontent.com/13646631/235270229-12424411-126c-440c-8487-1c187d09ce6d.JPG)

I can see where @calyptura 's suggestions would also help in this situation, but I expected Clear ALL data, to really clear this out too.  I did try re-loading the page--no effect.

**Reply by lloydbayley** [reply]
Did it show it for long? I'm wondering if that has to do with the Top 10 not updating immediately. The data may be gone but the top 10 hadn't updated yet? I've never really had cause to test that out. Also, were the services still running? If so, they would still be picking up birds and processing them, so I could be an intermediary thing that some were picked up after the Clear All ran and before you got check it.

**Reply by Astrobirder** [reply]
Yes, it persisted for a while--not sure how long.  Yes, all the services were still running, BUT I had unplugged the microphone BEFORE I did the clear all, and, as you can see BELOW the graph: "No Detections For Today".

**Reply by lloydbayley** [reply]
Well, gosh, I'm terribly sorry I didn't read the post properly.

**Reply by Astrobirder** [reply]
No worries @lloydbayley!  I just appreciate folks in the community trying to help each other and move the project along!

**Reply by lloydbayley** [reply]
I thought I was....anyway...there we go

**Reply by Astrobirder** [reply]
Oh, don't misunderstand me--you ARE!

**Comment by lloydbayley** [comment]
Both are not-bad ideas at all! Could be useful...I have thought of the delete on today's detections before. I'll have a look at it tomorrow (unless @ehpersonal38 gets there first) as it's 12:42am and I shouldn't be starting new stuff at this time! :)

**Reply by ehpersonal38** [reply]
I'll let you take this one! The initial reason I didn't do this is because on mobile there's not enough space for the buttons. But I think just the delete button should be alright.

EDIT: And if you really wanted to toast your pretzel roll you could add ALL the buttons and then put in some conditional css that hides them with `display: none` if the screen size is less than 700px or whatever mobile is.

**Reply by lloydbayley** [reply]
Jesus! Ya don't want much do ya? :)

**Reply by lloydbayley** [reply]
I've had a look into it and it's a rather complex thing to delete one. To be honest, I would be scared to try implementing that. Might have to defer to @ehpersonal38 . Sorry Ethan. Above my pay grade.

**Reply by ehpersonal38** [reply]
Heh, no worries! Luckily I already wrote the code a while ago and it just needed to be shoehorned in.

@calyptura I've added this now, you can update and check it out.

**Reply by lloydbayley** [reply]
Well now I'm even more glad I didn't battle with code! Sneaky old you having it written already and waiting for grasshopper to perform a submission as a test! I'll cut my teeth on some other code! :)

**Reply by Astrobirder** [reply]
Oh @ehpersonal38 IS sneaky!  AND I've already used the new function multiple times today--in my debugging efforts with the New model/Old Model.  I deleted the obviously incorrect IDs with this feature before I switched models.

**Reply by jflyons** [reply]
Liking that garbage can under Today's Detections - thanks !!!
